driver/layernorm_driver.hpp

```diff
@@ -0,0 +1,429 @@
+    int in_c = inflags.GetValueInt("in_channels");
+    int in_w = inflags.GetValueInt("in_w");
+    int in_h = inflags.GetValueInt("in_h");
+    int in_d = inflags.GetValueInt("in_d");
+
+    if(in_h != 0)
+    {
+        if(in_d != 0)
+        {
+            dim_size = 5;
+            return std::vector<int>({in_n, in_c, in_d, in_h, in_w});
+        }
+        else
+        {
+            dim_size = 4;
+            return std::vector<int>({in_n, in_c, in_h, in_w});
+        }
+    }
+    else
+    {
+        dim_size = 3;
+        return std::vector<int>({in_n, in_c, in_w});
+    }
+}
+
+template <typename Tgpu, typename Tref>
+int LayerNormDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t in_sz     = GetTensorSize(inputDesc);
+    size_t weight_sz = GetTensorSize(weightDesc);
+    size_t bias_sz   = GetTensorSize(biasDesc);
+    size_t out_sz    = GetTensorSize(outputDesc);
+    size_t mean_sz   = GetTensorSize(meanDesc);
+    size_t rstd_sz   = GetTensorSize(rstdDesc);
+
+    // MIOPEN_BACKEND_HIP
+    uint32_t ctx = 0;
+
+    in_dev     = std::unique_ptr<GPUMem>(new GPUMem(ctx, in_sz, sizeof(Tgpu)));
+    weight_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, weight_sz, sizeof(Tgpu)));
+    bias_dev   = std::unique_ptr<GPUMem>(new GPUMem(ctx, bias_sz, sizeof(Tgpu)));
+    out_dev    = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
+    mean_dev   = std::unique_ptr<GPUMem>(new GPUMem(ctx, mean_sz, sizeof(Tgpu)));
+    rstd_dev   = std::unique_ptr<GPUMem>(new GPUMem(ctx, rstd_sz, sizeof(Tgpu)));
+
+    in       = std::vector<Tgpu>(in_sz, static_cast<Tgpu>(0));
+    weight   = std::vector<Tgpu>(weight_sz, static_cast<Tgpu>(0));
+    bias     = std::vector<Tgpu>(bias_sz, static_cast<Tgpu>(0));
+    out      = std::vector<Tgpu>(out_sz, static_cast<Tgpu>(0));
+    mean     = std::vector<Tgpu>(mean_sz, static_cast<Tgpu>(0));
+    rstd     = std::vector<Tgpu>(rstd_sz, static_cast<Tgpu>(0));
+    outhost  = std::vector<Tref>(out_sz, static_cast<Tref>(0));
+    meanhost = std::vector<Tref>(mean_sz, static_cast<Tref>(0));
+    rstdhost = std::vector<Tref>(rstd_sz, static_cast<Tref>(0));
+
+    // MIOPEN_BACKEND_HIP
+    int status;
+
+    for(int i = 0; i < in_sz; i++)
+    {
+        in[i] = RAN_GEN<Tgpu>(static_cast<Tgpu>(0.0), static_cast<Tgpu>(1.0));
+    }
+    status = in_dev->ToGPU(q, in.data());
+
+    for(int i = 0; i < weight_sz; i++)
+    {
+        weight[i] = RAN_GEN<Tgpu>(static_cast<Tgpu>(0.0), static_cast<Tgpu>(1.0));
+    }
+    status = weight_dev->ToGPU(q, weight.data());
+
+    for(int i = 0; i < bias_sz; i++)
+    {
+        bias[i] = RAN_GEN<Tgpu>(static_cast<Tgpu>(0.0), static_cast<Tgpu>(1.0));
+    }
+    status = bias_dev->ToGPU(q, bias.data());
+
+    status |= out_dev->ToGPU(q, out.data());
+    status |= mean_dev->ToGPU(q, mean.data());
+    status |= rstd_dev->ToGPU(q, rstd.data());
+
+    if(status != CL_SUCCESS)
+        printf("Error copying data to GPU\n");
```

Comment:
[future][R] printf is a rudiment of ancient times 
```suggestion
    if(status != 0)
        std::cout <<  ... << std::endl
```
CL_SUCCESS is from OpenCL.